eventsource-client/src/client.rs

```diff
@@ -90,10 +109,18 @@ impl ClientBuilder {
     /// Build with a specific client connector.
     pub fn build_with_conn<C>(self, conn: C) -> impl Client
     where
-        C: Connect + Clone + Send + Sync + 'static,
+        C: Service<Uri> + Clone + Send + Sync + 'static,
```

Comment:
We have to stop using `Connect` here because `hyper_timeout` doesn't have a dependency on the `http1/2` features of hyper. See [here](https://github.com/hjr3/hyper-timeout/pull/20/files) for context.